PhysicsTools/Heppy/python/analyzers/objects/JetAnalyzer.py

```diff
@@ -119,12 +119,39 @@ def process(self, event):
         
 	##Sort Jets by pT 
         allJets.sort(key = lambda j : j.pt(), reverse = True)
+        
+        leptons = []
+        if hasattr(event, 'selectedLeptons'):
+            leptons = [ l for l in event.selectedLeptons if l.pt() > self.lepPtMin and self.lepSelCut(l) ]
+        if self.cfg_ana.cleanJetsFromTaus and hasattr(event, 'selectedTaus'):
+            leptons = leptons[:] + event.selectedTaus
+        if self.cfg_ana.cleanJetsFromIsoTracks and hasattr(event, 'selectedIsoCleanTrack'):
+            leptons = leptons[:] + event.selectedIsoCleanTrack
 	## Apply jet selection
         self.jets = []
         self.jetsFailId = []
         self.jetsAllNoID = []
         self.jetsIdOnly = []
         for jet in allJets:
+            #Check if lepton and jet have overlapping PF candidates 
+            leps_with_overlaps = []
+            for i in range(jet.numberOfSourceCandidatePtrs()):
+                p1 = jet.sourceCandidatePtr(i) #Ptr<Candidate> p1
+                if not p1.isAvailable():
+                    continue
+                for lep in leptons:
+                    for j in range(lep.numberOfSourceCandidatePtrs()):
+                        p2 = lep.sourceCandidatePtr(j)
+                        if not p2.isAvailable():
+                            continue
+                        #for some reason, this does not work.
+                        #has_overlaps = p1.refCore() == p2.refCore() and p1.key() == p2.key()
+                        has_overlaps = p1.p4() == p2.p4() and p1.charge() == p2.charge()
+                        if has_overlaps:
+                            leps_with_overlaps += [lep]
```

Comment:
can you explain the logic here?? while for each source candidate ptr (can be more than one for electrons) you add [lep] to leps_with_overlaps? i.e. add again the same lepton several time?